src/generators/generator.py

```diff
@@ -114,8 +114,20 @@ def gen_top_level_declaration(self):
             self.gen_class_decl,
```

Comment:
Ok let's do a quick trick here so that the core decl generators and the language-specific declarations are called in a similar manner.

```python
candidates = [
    lambda gen: gen.gen_variable_decl(),
    lambda gen: gen.gen_class_decl(),
    lambda gen: gen.gen_func_decl()
]
```

then you have

```python
gen_func = ut.random.choice(candidates)
gen_func(self)
```